lib/uncruft/railtie.rb

```diff
@@ -11,12 +11,16 @@ class Railtie < ::Rails::Railtie
         strategies.unshift(DeprecationHandler.new)
         config.active_support.deprecation = strategies
       end
+    end
 
-      if Rails.gem_version >= Gem::Version.new('7.1')
-        initializer "uncruft.deprecator" do |app|
-          app.deprecators[:uncruft] = Uncruft.deprecator
-        end
+    if Rails.gem_version >= Gem::Version.new('7.1')
+      initializer "uncruft.deprecator" do |app|
+        app.deprecators[:uncruft] = Uncruft.deprecator
       end
     end
+
+    initializer 'uncruft.deprecator.behavior' do |app|
+      Uncruft.deprecator.behavior = app.config.active_support.deprecation
+    end
```

Comment:
We need to inherit the deprecation behavior from the application.

I _really_ struggled to write a test for this behavior. I was able to verify that it's working, but I can't assert on the identity of `Uncruft.deprecator.behavior` because it's an anonymous proc that wraps the `Uncruft::DeprecationHandler`.

Also, this spec isn't resetting state properly.